substrate-node/support/src/resources.rs

```diff
@@ -79,11 +80,30 @@ impl Resources {
         let su = self.get_su();
         cu * 2 + su
     }
+
+    pub fn has_changed(&self, other: Resources, tolerance: u8) -> Result<(), ()> {
+        let wiggle = |a: u64, b: u64| -> Result<(), ()> {
+            let p = Percent::from_percent(tolerance) * a;
```

Comment:
Not familiar with the `Percent` function but seems to be good practice
https://github.com/paritytech/substrate/blob/8bc45ed/primitives/arithmetic/src/per_things.rs#L94